packages/components/src/components/as-range-slider/thumb/as-range-slider-thumb.tsx

```diff
@@ -122,8 +122,12 @@ export class RangeSliderThumb {
     const eventX = changedTouches ? changedTouches[0].pageX : (event as MouseEvent).pageX;
     const barPercentage = (eventX - this.railBoundingClientRect.left) * 100 / this.railElement.offsetWidth;
 
-    if (barPercentage < 0 || barPercentage > 100) {
-      return;
+    if (barPercentage < 0) {
+      return this.thumbMove.emit(0);
```

Comment:
My only minor concern with this is that if you'll get a lot of 'public' events thrown if you keep dragging after reaching the beginning or the end. I'd try to throttle it somehow on the parent component.

However, it's totally consistent with the current behavior for other steps. When dragging between two steps it fires n events, and not just one when the value changes, so I wouldn't worry too much about it. :man_shrugging:

Reply:
Yes, I tried to limit it by checking that percentage was different to 0 (`barPercentage < 0 && this.percentage !== 0`). But as percentage might not be exactly 0, but 0.0004 we need to use a "fuzzy" comparison to achieve that behaviour :(